content/articles/deploying-a-serverless-django-restapi-with-zappa-on-aws/index.md

```diff
@@ -0,0 +1,440 @@
+]
+```
+
+### Testing the API
+
+Lastly, test out the API to see the work we have done so far to ensure it's running successfully. To test locally, run this command in your terminal, python manage.py runserver, and in your browser hit the endpoint [http://120.0.0.1:8000/api/books](http://120.0.0.1:8000/api/books).
+
+***Hey! we are done building. Ladies and gentlemen let us deploy!***
+
+![Deployment](/engineering-education/deploying-a-serverless-jango-restapi-with-zappa-on-aws/meme.jpg)
+
+[Image source](https://www.google.com/url?sa=i&url=https%3A%2F%2Fmakeameme.org%2Fmeme%2Fbrace-yourself-deploy&psig=AOvVaw30-YKcrTN4IWmGoZeWsnAT&ust=1637270554714000&source=images&cd=vfe&ved=0CAsQjRxqFwoTCPiBt6GqoPQCFQAAAAAdAAAAABAJ)
+
+
+### Setting up AWS
+
+At this point, it is supposed that you have created your AWS account and have access to an AWS IAM user for the Zappa utility. Zappa utility handles tasks such as automatically creating an s3 bucket for deployment, creating lambda execution IAM roles, API gateways for web traffic work, etc. on your behalf.
+
+Briefly, let us set up AWS on your console to properly utilize Zappa.
+
+S*tep 1*: Log in to your AWS console and go to the IAM section
+
+*Step 2*: On the left side of the screen click on *users*
+
+*Step 3*: Click on *add users*. For this project, we will name our 'user', *'serverless'* and for access type, choose '*Programmatic access*'. This uses the API to interact with AWS.
+
+*Step 4*: Move on the permission and select *'Attach existing policies directly*. We will be accessing two policies, the first is '*IAMFullAccess*" This creates users for lambda to be executed. The second is *'PowerUserAccess'* for creating API gateway and S3 bucket. Search for these policies and click on their boxes.
+
+*Step 5*: Skip Add tags, move to Review and create a user.
+
+*Successfully created!* We have also been supplied with an access key and secret access key. Download the *.csv file* which contains these keys as these pieces of information will be lost when the page closes
+
+Briefly, let'sSet up AWS on your console in other to properly utilize Zappa. 
+
+### Deploying with Zappa
+
+The first step is to install Zappa in our virtual environment:
+
+```python
+pip install zappa
+```
+
+Next, initialize Zappa, 
+
+```python
+zapa init
+```
+
+```python
+███████╗ █████╗ ██████╗ ██████╗  █████╗
+╚══███╔╝██╔══██╗██╔══██╗██╔══██╗██╔══██╗
+  ███╔╝ ███████║██████╔╝██████╔╝███████║
+ ███╔╝  ██╔══██║██╔═══╝ ██╔═══╝ ██╔══██║
+███████╗██║  ██║██║     ██║     ██║  ██║
+╚══════╝╚═╝  ╚═╝╚═╝     ╚═╝     ╚═╝  ╚═╝
+
+Welcome to Zappa!
+
+Zappa is a system for running server-less Python web applications on AWS Lambda and AWS API Gateway.
+This `init` command will help you create and configure your new Zappa deployment.
+Let's get started!
+
+Your Zappa configuration can support multiple production stages, like 'dev', 'staging', and 'production'.
+What do you want to call this environment (default 'dev'): dev
+```

```

Comment:
Can we add image for this?

Not sure, if this would be rendered properly

Reply:
Images in place of codes?

Reply:
Not all the codes. Just this block.

Reply:
Done.

Reply:
The above code block can be removed, since we have the image already